wadas/ui/configure_actuators_dialog.py

```diff
@@ -362,24 +361,29 @@ def start_actuator_server(self):
                 self.ui.label_cert_file.text(),
                 self.ui.label_key_file.text(),
             )
-
+        self._setup_logger()
         self.ui.pushButton_stop_server.setEnabled(True)
         # Start the thread
         self.actuator_server_thread = self.actuator_server.run()
+        self.ui.pushButton_start_server.setEnabled(False)
 
     def stop_actuator_server(self):
         """Method to stop the Actuator server."""
 
         if self.actuator_server and self.actuator_server_thread:
             self.actuator_server.stop()
             self.actuator_server_thread.join()
-            self.ui.pushButton_stop_server.setEnabled(False)
+
+        self.ui.pushButton_stop_server.setEnabled(False)
+        self.ui.pushButton_start_server.setEnabled(True)
 
     def _setup_logger(self):
         """Initialize logger for UI logging."""
-
-        # TODO: fix log redirecting to UI dialog only
-        logger = logging.getLogger("fastapi")
+        logger_names = ["uvicorn", "uvicorn.error", "uvicorn.access"]
         log_textbox = QTextEditLogger(self.ui.plainTextEdit_test_server_log)
-        logger.setLevel(logging.DEBUG)
-        logger.addHandler(log_textbox)
+        for logger_name in logger_names:
+            server_logger = logging.getLogger(logger_name)
+            for handler in server_logger.handlers[:]:
+                server_logger.removeHandler(handler)
+            server_logger.addHandler(log_textbox)
+            server_logger.propagate = False
```

Comment:
This is very similar to L22-30 in fastapi_actuator_server.py. You may create a common function that just takes a couple of arguments: the handler to add and an optional debug level.